pkg/registry/core/shoot/strategy.go

```diff
@@ -166,6 +171,25 @@ func removeForbiddenFinalizers(shoot *core.Shoot) {
 	shoot.Finalizers = finalizers
 }
 
+func removeDuplicateExtension(shoot *core.Shoot) {
+	if len(shoot.Spec.Extensions) > 1 {
+		extensionsType := map[string]core.Extension{}
```

Comment:
```suggestion
		extensionsType := make(map[string]core.Extension, len(shoot.Spec.Extensions))
```
Maybe call it `typeToExtension` map, or set?